lib/rbs/cli.rb

```diff
@@ -750,5 +751,55 @@ def parse_type_name(string)
         TypeName.new(name: last, namespace: namespace.parent)
       end
     end
+
+    def test_opt options
+      opt_string = options.dirs.map { |dir| "-I #{dir}"}.concat(options.libs.map { |lib| "-r{lib}"}).join(' ')
+      opt_string.empty? ? nil : Shellwords.escape(opt_string)
+    end
+
+    def run_test(args, options)
+      targets = []
+      sample_size = nil
+      help = ""
+
+      OptionParser.new do |opts|
+        opts.banner = <<EOB
+Usage: rbs [rbs options...] test [test options...] COMMAND
+
+Examples:
+
+  $ rbs test rake test
+  $ rbs --log-level=debug test --target SomeModule::* rspec
+  $ rbs test --target SomeModule::* --target AnotherModule::* --target SomeClass rake test
+
+Options:
+EOB
+        opts.on("--target TARGET", "Sets the runtime test target") do |target|
+          targets << Shellwords.escape(target)
+        end
+
+        opts.on("--sample-size SAMPLE_SIZE", "Sets the sample size") do |size|
+          sample_size = size
+        end
+
+        help = opts.help
+      end.order!(args)
+
+      if args.length.zero?
+        stdout.puts help
+        exit 1
+      end
+
+      targets_string = targets.uniq.join(',')
+
+      env_hash = {
+        'RBS_TEST_OPT' => test_opt(options),
+        'RBS_TEST_TARGET' => (targets_string unless targets_string.empty?),
```

Comment:
I would write `targets.join(",") unless targets.empty?` here.

* It works well even without `uniq`.
* Testing `targets` emptiness seems better because we can omit `targets_string` variable completely.